google/cloud/opentelemetry/internal/recordable.cc

```diff
@@ -28,6 +28,29 @@ namespace {
 // http://35.193.25.4/TensorFlow/models/research/syntaxnet/util/utf8/unilib_utf8_utils.h
 bool IsTrailByte(char x) { return static_cast<signed char>(x) < -0x40; }
 
+// OpenTelemetry's semantic conventions for attribute keys differ from Cloud
+// Trace's semantics for label keys. So translate from one to the other.
+//
+// See: https://cloud.google.com/trace/docs/trace-labels#canonical_labels
+opentelemetry::nostd::string_view MapKey(
+    opentelemetry::nostd::string_view key) {
+  static auto const* m = new std::map<std::string, std::string>{
+      {"http.host", "/http/host"},
+      {"http.method", "/http/method"},
+      {"http.target", "/http/path"},
+      {"http.status_code", "/http/status_code"},
+      {"http.url", "/http/url"},
+      {"http.user_agent", "/http/user_agent"},
+      {"http.request_content_length", "/http/request/size"},
+      {"http.response_content_length", "/http/response/size"},
+      {"http.scheme", "/http/client_protocol"},
+      {"http.route", "/http/route"},
+  };
+  auto it = m->find({key.data(), key.size()});
+  if (it == m->end()) return key;
```

Comment:
Observation:  It doesn't matter in the one call to this function (because the result it assigned to the argument), but a `string_view` returning function usually has to make some statement about the lifetime of its result.

Reply:
Changed the function to just modify a `string_view&`.

I assume the data in that `static auto const* m` thing lasts forever.

Reply:
Yes ... the `"..."` string literals are valid for the entire program.

I guess the new signature is a little clearer in that "no change" is now really no change (instead of self assignment).  The function should probably still say that it would only extend the lifetime, but perhaps that is now clear without further statement (and it is only file local).

Reply:
I liked the original signature better 🤷